### PR TITLE
Correct rollout status behavior when watching multiple resources

### LINKED ISSUE #1050: rollout status not return a message
#### What happened:

The client don't return a message. After use the command.
```
kubectl rollout status deployment 

```
#### What you expected to happen:
I expected the same behavior of the other commands of rollout. For example
```
kubectl rollout history deployment 
deployment.apps/nginx191 
REVISION  CHANGE-CAUSE
1         <none>
``` 

#### How to reproduce it (as minimally and precisely as possible):
```
kubectl rollout status deployment 

```
#### Environment:
- Kubernetes version (use `kubectl version`): 
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T21:16:14Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-21T01:11:42Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by rikatz**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
kubectl rollout status is not watching resource collections correctly. It keeps stuck without any response.

This PR corrects this behavior, while using "watch=true" now being required to watch a collection of resources
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/1050

#### Does this PR introduce a user-facing change?
```release-note
kubectl rollout now can watch multiple resources
```

/sig cli
/priority backlog
/area kubectl

### CODE REVIEW

**Comment by rikatz** [comment]
/cc @dougsland @lauchokyip 

Folks, if you can take a look I would really appreciate :)

**Comment by lauchokyip** [comment]
will try 😄

**Review by lauchokyip — COMMENTED** [review]
Hey @rikatz , tried my best and left some reviews :)

Also since the `kubectl rollout status TYPE` will trigger the watch, should you change the description on [here](https://github.com/rikatz/kubernetes/blob/ea5443356e56784cc17d98736b5ac177d9c5e5d6/staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go#L49-L54) too ?

File: staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go [hunk 1]
```diff
@@ -154,6 +154,10 @@ func (o *RolloutStatusOptions) Validate() error {
 		return fmt.Errorf("revision must be a positive integer: %v", o.Revision)
 	}
 
+	if len(o.BuilderArgs) == 1 && !o.Watch {
+		return fmt.Errorf("Watch must be true when watching a resource collection")
```
[the remark this answers is not in the document]
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_status.go:158]: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Pull request closed without merging**